DownFall/Views/ParagraphNode.swift

```diff
@@ -0,0 +1,137 @@
+    }
+    
+    func fontColor(newValue: UIColor) {
+        self.fontColor = newValue
+        retexture()
+    }
+
+    func text(newValue: String) {
+        self.text = newValue
+        retexture()
+    }
+    
+    func fontSize(newValue: CGFloat) {
+        self.fontSize = newValue
+        retexture()
+    }
+    
+    func paragraphWidth(newValue: CGFloat) {
+        self.paragraphWidth = newValue
+        retexture()
+    }
+    
+    init(text: String,
+         paragraphWidth: CGFloat,
+         fontName: String = ParagraphNode.defaultFontName,
+         fontSize: CGFloat = ParagraphNode.defaultFontSize,
+         fontColor: UIColor = ParagraphNode.defaultFontColor
+         ) {
+        self.fontName = fontName
+        self.text = text
+        self.fontSize = fontSize
+        self.fontColor = fontColor
+        self.paragraphWidth = paragraphWidth - 2 * Style.paragraphPadding
+        super.init(texture: nil, color: .clear, size: .zero)
+        
+        retexture()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    static func labelNode(text: String,
+                          paragraphWidth: CGFloat,
+                          fontName: String = ParagraphNode.defaultFontName,
+                          fontSize: CGFloat = ParagraphNode.defaultFontSize,
+                          fontColor: UIColor = ParagraphNode.defaultFontColor
+                          ) -> ParagraphNode {
+        return ParagraphNode(
+            text: text,
+            paragraphWidth: paragraphWidth,
+            fontName: fontName,
+            fontSize: fontSize,
+            fontColor: fontColor
+        )
+    }
+    
+    func retexture() {
+        guard let image = imageFrom(text: self.text as NSString) else { return }
+        let texture = SKTexture(image: image)
+        self.texture = texture
+        self.anchorPoint = CGPoint(x:0.5, y:0.5)
+    }
+    
+    func imageFrom(text: NSString) -> UIImage? {
+        
+        // paragraph styling
+        let paragraphStyle = NSMutableParagraphStyle()
+        paragraphStyle.lineBreakMode = NSLineBreakMode.byWordWrapping
+        paragraphStyle.alignment = NSTextAlignment.left
+        paragraphStyle.lineSpacing = 1;
+
+        // attributes
+        var attributes: [NSAttributedString.Key: Any] = [:]
+        guard let customFont = UIFont(name: "Alterebro-Pixel-Font", size: fontSize) else {
+            fatalError("""
+                Failed to load the "pixel-font" font.
+                Make sure the font file is included in the project and the font name is spelled correctly.
+                """
+            )
+        }
+
+        attributes[NSAttributedString.Key.font] = customFont
+        attributes[NSAttributedString.Key.paragraphStyle] = paragraphStyle
+        attributes[NSAttributedString.Key.foregroundColor] = fontColor
+
+
+        let size = CGSize(width: paragraphWidth, height: 2000.0)
```

Comment:
Can we program the height? And would it be possible to make the font shrink if the string at a certain size doesn't fit?  Could we make it shrink a certain amount before cutting off words with a `...`

Reply:
I added a trello